stl/inc/mdspan

```diff
@@ -228,6 +226,14 @@ public:
             }
         }
     }
+
+    _NODISCARD static constexpr bool _Is_index_space_size_representable() {
+        if constexpr (rank_dynamic() == 0 && rank() > 0) {
+            return _STD in_range<index_type>((_Extents * ...));

```

Comment:
Do we have to worry about integer overflow during this monster multiplication?

Reply:
The only references I could find to `index_type` being able to represent the index space size are in *Preconditions* to the various layout mapping constructors. If we're going to upgrade it to a `static_assert`, then I think we do need to check for overflow, because it can happen and the compiler is supposed to diagnose UB in constant evaluation contexts. We also don't want any false positives, since the user won't have any way to bypass this check.